iox_query/src/exec/seriesset/converter.rs

```diff
@@ -491,32 +502,29 @@ impl Stream for SeriesSetConverterStream {
 #[derive(Debug)]
 pub struct GroupGenerator {
     group_columns: Vec<Arc<str>>,
+    memory_manager: Arc<MemoryManager>,
 }
 
 impl GroupGenerator {
-    pub fn new(group_columns: Vec<Arc<str>>) -> Self {
-        Self { group_columns }
+    pub fn new(group_columns: Vec<Arc<str>>, memory_manager: Arc<MemoryManager>) -> Self {
+        Self {
+            group_columns,
+            memory_manager,
+        }
     }
 
     /// groups the set of `series` into SeriesOrGroups
     ///
     /// TODO: make this truly stream-based
```

Comment:
```suggestion
    /// TODO: make this truly stream-based
    /// https://github.com/influxdata/influxdb_iox/issues/6347
```

I filed a ticket to track the truly streaming solution. I don't think we should pursue it unless we get people explicitly asking for it / hitting this ResourcesExhausted limit